test/src/core/testCreateSession.js

```diff
@@ -432,4 +432,255 @@ describe('test Request With Credentials', function () {
       sessionId: '1352110219202'
     });
   });
+
+  it('Test create session with browserstack', async function() {
+    nock('https://hub.browserstack.com')
+      .post('/wd/hub/session')
+      .reply(201, function (uri, requestBody) {
+        const reqObj = JSON.parse(requestBody);
+
+        assert.deepEqual(reqObj, {
+          'desiredCapabilities': {
+            'browserName': 'chrome',
+            'bstack:options': {
+              'local': 'false',
+              'userName': 'test_user',
+              'accessKey': 'test_key',
+              'os': 'OS X',
+              'osVersion': 'Monterey',
+              'buildName': 'nightwatch-test-build'
+            },
+            'build': 'Nightwatch Programmatic Api Demo',
+            'chromeOptions': {
+              'w3c': false
+            },
+            'goog:chromeOptions': {
+              'w3c': false
+            },
+            'name': 'Try 1'
+          },
+          'capabilities': {
+            'alwaysMatch': {
+              'browserName': 'chrome',
+              'bstack:options': {
+                'local': 'false',
+                'userName': 'test_user',
+                'accessKey': 'test_key',
+                'os': 'OS X',
+                'osVersion': 'Monterey',
+                'buildName': 'nightwatch-test-build'
+              },
+              'goog:chromeOptions': {
+                'w3c': false
+              }
+            }
+          }
+        });
+
+        return {
+          value: {
+            sessionId: '1352110219202',
+            capabilities: reqObj.desiredCapabilities
+          }
+        };
+      });
+
+
+    nock('https://api.browserstack.com')
+      .get('/automate/builds.json')
+      .reply(200,  [
+        {
+          automation_build: {
+            name: 'WIN_CHROME_PROD_SANITY_LIVE_1831',
+            duration: 47,
+            status: 'running',
+            hashed_id: '8dd73aad3365429dec0ec12cf64c0c475a22dasds',
+            build_tag: null
+          }
+        },
+        {
+          automation_build: {
+            name: 'External monitoring - aps - 2022-08-30',
+            duration: 44,
+            status: 'done',
+            hashed_id: '8dd73aad3365429dec0ec12cf64c0c475a22dasdk',
+            build_tag: null
+          }
+        } 
+      ]);
+    
+    const client = Nightwatch.createClient({
+      webdriver: {
+        start_process: false
+      },
+      selenium: {
+        host: 'hub.browserstack.com',
+        port: 443
+      },
+      desiredCapabilities: {
+        'bstack:options': {
+          local: 'false',
+          userName: 'test_user',
+          accessKey: 'test_key',
+          os: 'OS X',
+          osVersion: 'Monterey'
+        },
+        browserName: 'chrome',
+        chromeOptions: {
+          w3c: false
+        }
+      },
+
+      parallel: false
+    });
+
+    client.mergeCapabilities({
+      name: 'Try 1',
+      build: 'Nightwatch Programmatic Api Demo'
```

Comment:
we should try updating in `buildName` inside `bstack:options` as that is used according w3c protocol: https://www.browserstack.com/automate/capabilities?tag=selenium-4

Reply:
Some observations for future reference:
| desiredCapabilities | bstack:options | result |
| -------------------| --------------- | ------ |
|`buildName` | `buildName`  | takes value from `buildName` of `bstack:options`|
|`build`|`buildName`| takes value from `build` of `desiredCapabilities`| 
|`buildName` | `undefined`  | takes value from `buildName` of  `desiredCapabilities`|
|`build`|`undefined`| takes value from `build` of `desiredCapabilities`| 
|`build` and `buildName` | `undefined` | takes value from `build` of `desiredCapabilities`|

Reply:
resolved

Reply:
Shouldn't it take `buildName` from `bstack:options` when `build` and `buildName` is present. As that is the newer one according to w3c protcol

Reply:
@gravityvi the above observations is the current behaviour of Browserstack Automate, not related to nightwatch.